File: src/_1ProgrammingBasicsWithJavaMay2023/_2ConditionalStatements/_2Exercise/_06WorldSwimmingRecord.java
package _1ProgrammingBasicsWithJavaMay2023._2ConditionalStatements._2Exercise;

import java.util.Scanner;

public class _06WorldSwimmingRecord {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double record = Double.parseDouble(scanner.nextLine());
        double distance = Double.parseDouble(scanner.nextLine());
        double timePer1Meter = Double.parseDouble(scanner.nextLine());
        double timeForSwim = distance * timePer1Meter;
        double sectorsInDistance = Math.floor(distance / 15);
        double slowingTime= sectorsInDistance * 12.5;
        timeForSwim = timeForSwim + slowingTime;
        double diff = timeForSwim - record;
        if (timeForSwim < record) {
            System.out.printf("Yes, he succeeded! The new world record is %.2f seconds.", timeForSwim);
        } else {
            System.out.printf("No, he failed! He was %.2f seconds slower.", diff);
        }
    }
}

File: src/_1ProgrammingBasicsWithJavaScriptNovember2024/_2ConditionalStatements/_2Exercise/_06WorldSwimmingRecord.js
function worldSwimmingRecord(recordInSeconds, distanceInMeters, metersInSecond) {
    const SLOWER_FOR_EVERY_15_METERS = 12.5;

    const slowTimes = Math.floor(distanceInMeters / 15);
    const secondsForNormalSwim = distanceInMeters * metersInSecond;
    const secondsSlow = slowTimes * SLOWER_FOR_EVERY_15_METERS;
    const totalTime = secondsForNormalSwim + secondsSlow;

    if (totalTime < recordInSeconds) {
        console.log(`Yes, he succeeded! The new world record is ${totalTime.toFixed(2)} seconds.`);
    } else {
        const secondsSlowerToBeatTheRecord = totalTime - recordInSeconds;
        console.log(`No, he failed! He was ${secondsSlowerToBeatTheRecord.toFixed(2)} seconds slower.`);
    }
}